### PR TITLE
BloodGroup example from Data Connect specification

### PR DESCRIPTION
**Pull request by jfuerth**
This type has been mentioned in the Data Connect (formerly Seach) specification for some time. Adding it here as suggested in https://github.com/ga4gh-discovery/data-connect/issues/96.

### CODE REVIEW

File: schemas/BloodGroup.yaml [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+      id: "orcid:0000-0001-9135-2164"
+  used_by:
+    - description: "Example schema in Data Connect specification"
+  sb_status: proposed
+description: |
+  An OntologyReference constrained to ABO Blood Groups
+allOf:
+  - "$ref": "https://schemablocks.org/schemas/sb-phenopackets/v1.0.0/OntologyClass.json#/properties"
+  - properties:
+      id:
```
Comment (mbaudis) [inline schemas/BloodGroup.yaml:17]: Hmmm... IMO the correct way would be to reference https://schemablocks.org/schemas/sb-phenopackets/current/OntologyClass.json#properties So you would have

```
type: array
items:
  $ref: "https://schemablocks.org/schemas/sb-phenopackets/current/OntologyClass.json#properties"
```
... and then have a note about the option in the description. But this, actually, shows the shortcomings of the subtractive model in JSON Schema where it is (AFAIK) harder then to both recycle an object, but specify its values; your direct definition is more compact...

A testing middle-way would be to use your definition, but with the `OntologyClass` `id`, `label` syntax. But I'm no specialist on JSON Schema...

**Review by mbaudis — APPROVED** [review]
Just merging, since playground & proposed status & discussions needed.

**Comment by mbaudis** [comment]
Mostly to challenge & test. Low-level status.

File: schemas/BloodGroup.yaml (hunk 1 above)
Comment (jfuerth) [reply schemas/BloodGroup.yaml:17]: Thanks for the review and feedback! The model I submitted was meant to do these things:

1. Specify that the attribute is an OntologyClass
2. Accept OntologyClass objects that are HPO BloodGroups, like `{ "id": "HP:0032442", "label": "O"}`
3. Reject OntologyClass objects that are not HPO BloodGroups, like `{ "id": "HP:0040068", "label": "Abnormality of limb bone"}`

Under the `id` property, the `allOf` array with `const` & `title` entries is just an alternative way of modelling an enum while being able to label each enumerated constant. JSON Schema's `enum` doesn't allow attaching any kind of description to each enum constant, and [this is apparently the recommended workaround](https://github.com/json-schema-org/json-schema-spec/issues/57#issuecomment-247861695).

I'm definitely not sure I did this the best way, or even in a good way :). Just wanted to follow up to clarify my intent.

I would love to be able to specify "this property is an OntologyClass which is a reachable subclass of `HP:0032224` (ABO blood group)"... maybe we could come up with a technique for modelling this. Being able to link JSON Schemas to existing OWL ontologies could be really powerful.

**Pull request merged**